src/utils/reducers/queueInterval.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * Used to "findIndex" by uuid in an array.
+ */
+const findByUuid = function(uuid) {
+  return (o => o.uuid === uuid);
+}
+
+
+/*
+ * Used to "filter" out by uuid in an array.
+ */
+const filterByUuid = function(uuid) {
+  return (o => o.uuid !== uuid);
+}
+
+
+/*
+ * Used to "sort" and array containing queue intervals.
+ */
+const sortQueueIntervals = function(a, b) {
+  if(!b.upperBound) {
+    return -1;
+  } else if(!a.upperBound) {
+    return 1;
+  } else {
+    return (
+      a.upperBound.timestampMilliseconds -
+      a.upperBound.timestampMilliseconds
+    );
+  }
+}
+
+
+/*
+ * Create a queue interval relevant to a queue item (track).
+ */
+export const queueIntervalCreate = function(state, action) {
```

Comment:
these are kind of dense, but if you read the comments it should make sense. you may need to review the UML diagram/ serialization methods on the backend to fully comprehend the expected data structure.